src/gp_activetable.c

```diff
@@ -198,15 +202,78 @@ object_access_hook_QuotaStmt(ObjectAccessType access, Oid classId, Oid objectId,
 {
 	if (prev_object_access_hook) (*prev_object_access_hook)(access, classId, objectId, subId, arg);
 
-	/* if is 'drop extension diskquota' */
-	if (classId == ExtensionRelationId && access == OAT_DROP)
+	if (access == OAT_DROP)
 	{
-		if (get_extension_oid("diskquota", true) == objectId)
+		switch (classId)
 		{
-			invalidate_database_rejectmap(MyDatabaseId);
-			diskquota_stop_worker();
+			case ExtensionRelationId:
+				/* if is 'drop extension diskquota' */
+				if (get_extension_oid("diskquota", true) == objectId)
+				{
+					invalidate_database_rejectmap(MyDatabaseId);
+					diskquota_stop_worker();
+				}
+				return;
+
+			case NamespaceRelationId:
+			case AuthIdRelationId:
+			case TableSpaceRelationId: {
+				if (Gp_role != GP_ROLE_DISPATCH) return;
+
+				Oid namespaceId = get_namespace_oid("diskquota", true);
+				if (!OidIsValid(namespaceId) || !OidIsValid(get_relname_relid("quota_config", namespaceId)) ||
+				    !OidIsValid(get_relname_relid("target", namespaceId)))
+					return;
+
+				bool connected_in_this_function = SPI_connect_if_not_yet();
+				if (classId == TableSpaceRelationId)
+				{
+					SPI_execute_with_args("delete from diskquota.target where tablespaceOid = $1", 1,
+					                      (Oid[]){
+					                              OIDOID,
+					                      },
+					                      (Datum[]){
+					                              ObjectIdGetDatum(objectId),
+					                      },
+					                      NULL, false, 0);
+				}
+				else
+				{
+					SPI_execute_with_args(
+					        "delete from diskquota.target where quotaType = $1 and primaryOid = $2", 2,
+					        (Oid[]){
+					                INT4OID,
+					                OIDOID,
+					        },
+					        (Datum[]){
+					                Int32GetDatum(classId == NamespaceRelationId ? NAMESPACE_TABLESPACE_QUOTA
+					                                                             : ROLE_TABLESPACE_QUOTA),
+					                ObjectIdGetDatum(objectId),
+					        },
+					        NULL, false, 0);
+				}
+
+				SPI_execute_with_args(
+				        "delete from diskquota.quota_config where (quotaType = $1 and targetOid = $2)"
+				        " or (quotaType in (2, 3) and (targetOid, quotaType) not in (select rowId, quotaType from "
+				        "diskquota.target))",
+				        2,
+				        (Oid[]){
+				                INT4OID,
+				                OIDOID,
+				        },
+				        (Datum[]){
+				                Int32GetDatum(classId == NamespaceRelationId
+				                                      ? NAMESPACE_QUOTA
+				                                      : (classId == AuthIdRelationId ? ROLE_QUOTA : TABLESPACE_QUOTA)),
+				                ObjectIdGetDatum(objectId),
+				        },
+				        NULL, false, 0);
+
+				SPI_finish_if(connected_in_this_function);
+				return;
```

Comment:
before patch:
if `classId` in (`NamespaceRelationId`, `AuthIdRelationId`, `TableSpaceRelationId`) and `access` == `OAT_DROP` then we don't return here
after patch:
with same condition we return here
why?

Reply:
before the patch we returned here
```
	if (classId != RelationRelationId || subId != 0)
	{
		return;
	}
```
So the behavior is the same